File: TextosJapones/src/main/java/org/jisho/textosJapones/controller/PopupNotificacaoController.java
package org.jisho.textosJapones.controller;

import java.net.URL;
import java.util.ResourceBundle;

import javafx.fxml.FXML;
import javafx.fxml.Initializable;
import javafx.scene.control.Label;
import javafx.scene.image.Image;
import javafx.scene.image.ImageView;
import javafx.scene.layout.AnchorPane;

public class PopupNotificacaoController implements Initializable {

	@FXML
	private AnchorPane notificacaoBackground;

	@FXML
	private ImageView imgImagem;

	@FXML
	private Label lblTitulo;

	@FXML
	private Label lblTexto;

	public Double wheight;

	public Image getImagem() {
		return imgImagem.getImage();
	}

	public PopupNotificacaoController setImagem(ImageView imagem) {
		this.imgImagem = imagem;
		return this;
	}

	public String getTitulo() {
		return lblTitulo.getText();
	}

	public PopupNotificacaoController setTitulo(String titulo) {
		this.lblTitulo.setText(titulo);
		return this;
	}

	public String getTexto() {
		return lblTexto.getText();
	}

	public PopupNotificacaoController setTexto(String texto) {
		lblTexto.setText(texto);

		if (texto.length() <= 80) {
			notificacaoBackground.setPrefSize(500, 47);
			notificacaoBackground.setMaxSize(500, 47);
			imgImagem.setFitHeight(35);
			imgImagem.setFitWidth(35);
			wheight = 60.0;
		} else {
			if (texto.length() <= 225) {
				notificacaoBackground.setPrefSize(500, 80);
				notificacaoBackground.setMaxSize(500, 80);
				imgImagem.setFitHeight(35);
				imgImagem.setFitWidth(35);
				wheight = 80.0;
			} else {
				notificacaoBackground.setPrefSize(650, 100);
				notificacaoBackground.setMaxSize(650, 100);
				imgImagem.setFitHeight(45);
				imgImagem.setFitWidth(45);
				wheight = 100.0;
			}
		}
		return this;
	}

	@Override
	public void initialize(URL location, ResourceBundle resources) {

	}
	
	public static URL getFxmlLocate() {
		return PopupNotificacaoController.class
				.getResource("/view/PopupNotificacao.fxml");
	}
}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/controller/PopupNotificacaoController.kt
package br.com.fenix.processatexto.controller

import javafx.fxml.FXML
import javafx.fxml.Initializable
import javafx.scene.control.Label
import javafx.scene.image.Image
import javafx.scene.image.ImageView
import javafx.scene.layout.AnchorPane
import java.net.URL
import java.util.*


class PopupNotificacaoController : Initializable {

    @FXML
    private lateinit var notificacaoBackground: AnchorPane

    @FXML
    private lateinit var imgImagem: ImageView

    @FXML
    private lateinit var lblTitulo: Label

    @FXML
    private lateinit var lblTexto: Label

    var wheight: Double = 0.0
    val imagem: Image get() = imgImagem.image

    fun setImagem(imagem: ImageView): PopupNotificacaoController {
        imgImagem = imagem
        return this
    }

    val titulo: String get() = lblTitulo.text

    fun setTitulo(titulo: String): PopupNotificacaoController {
        lblTitulo.text = titulo
        return this
    }

    val texto: String
        get() = lblTexto.text

    fun setTexto(texto: String): PopupNotificacaoController {
        lblTexto.text = texto
        wheight = if (texto.length <= 80) {
            notificacaoBackground.setPrefSize(500.0, 47.0)
            notificacaoBackground.setMaxSize(500.0, 47.0)
            imgImagem.fitHeight = 35.0
            imgImagem.fitWidth = 35.0
            60.0
        } else {
            if (texto.length <= 225) {
                notificacaoBackground.setPrefSize(500.0, 80.0)
                notificacaoBackground.setMaxSize(500.0, 80.0)
                imgImagem.fitHeight = 35.0
                imgImagem.fitWidth = 35.0
                80.0
            } else {
                notificacaoBackground.setPrefSize(650.0, 100.0)
                notificacaoBackground.setMaxSize(650.0, 100.0)
                imgImagem.fitHeight = 45.0
                imgImagem.fitWidth = 45.0
                100.0
            }
        }
        return this
    }

    @Override
    override fun initialize(location: URL, resources: ResourceBundle) {
    }

    companion object {
        val fxmlLocate: URL get() = PopupNotificacaoController::class.java.getResource("/view/PopupNotificacao.fxml")
    }
}